tools/gcp/utils/create_matrix_images.py

```diff
@@ -28,25 +28,39 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-"""Upload docker images to Google Container Registry."""
+"""Build and upload docker images to Google Container Registry per matrix."""
```

Comment:
tools/gcp/utils  (read as "utilities for Google computing platform") is a wrong place for these scripts IMHO.
Do you want to create  e.g. tools/interop_matrix  (or a different name, but just "matrix" is not specific enough) where all these scripts will live?  You can add a tools/interop_matrix/README.md that explains what you are trying to achieve.